### PR TITLE
feat: autodiscover repositories by namespace

### LINKED ISSUE #12635: Gitlab - autodiscover by groups
### What would you like Renovate to be able to do?

When using `autodiscover` (https://github.com/renovatebot/renovate/blob/main/docs/usage/self-hosted-configuration.md#autodiscover), Renovate retrieve all repositories that the bot account can access.  
On my Gitlab Hosted instances, there is thousands of projetcs, and it takes long time to retrieve all theses projects.  
I want to be able to retrieve projects only from a list of Gitlab groups.

### If you have any ideas on how this should be implemented, please tell us here.

For Gitlab, use the group API: https://docs.gitlab.com/ee/api/groups.html#list-a-groups-projects (`GET /groups/:id/projects` with parameter `include_subgroups=true`)

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by fgreinacher**
## Changes

Adds a new configuration option `autodiscoverNamespaces` to run Renovate against all repositories in a namespace.

## Context

Closes #12635

/cc @nejch 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

:hammer_and_pick: with :heart: by [Siemens](https://opensource.siemens.com/)

### CODE REVIEW

File: lib/modules/platform/gitlab/index.ts [hunk 1]
```diff
@@ -162,21 +162,43 @@ export async function getRepos(config?: AutodiscoverConfig): Promise<string[]> {
...
+  }
 
   try {
-    const res = await gitlabApi.getJson<RepoResponse[]>(url, {
-      paginate: true,
-    });
-    logger.debug(`Discovered ${res.body.length} project(s)`);
-    return res.body
+    const repos = (
+      await Promise.all(
```
Comment (viceice) [inline lib/modules/platform/gitlab/index.ts:183]: use pmap util to limit concurrency

Comment (fgreinacher) [reply lib/modules/platform/gitlab/index.ts]: Ah nice! Done with 
4d1be2e

File: lib/modules/platform/gitlab/index.ts [hunk 2]
```diff
@@ -162,21 +163,46 @@ export async function getRepos(config?: AutodiscoverConfig): Promise<string[]> {
...
-    return res.body
+    const repos = (
+      await pMap(
+        urls,
+        (url) =>
+          gitlabApi.getJson<RepoResponse[]>(url, {
+            paginate: true,
+          }),
+        {
+          concurrency: 2,
```
Comment (fgreinacher) [inline lib/modules/platform/gitlab/index.ts:191]: Could also pick a higher number here, but 2 felt fair and will usually be enough. I expect most folks to filter on just one namespace anyway.

**Review by viceice — APPROVED** [review]

**Review by secustor — APPROVED** [review]

**Review by jonesk7734 — COMMENTED** [review]
Ok

**Pull request merged**